aicsimageio/tests/readers/test_array_like_reader.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import dask.array as da
+import numpy as np
+import pytest
+
+from aicsimageio import dimensions, exceptions
+from aicsimageio.readers import ArrayLikeReader
+
+from ..image_container_test_utils import run_image_container_checks
+
+
+@pytest.mark.parametrize(
+    "array_constructor",
+    [
+        np.ones,
+        da.ones,
+    ],
+)
+@pytest.mark.parametrize(
+    "arr_shape, known_dims, expected_shape, expected_dims, expected_channel_names",
+    [
+        ((1, 1), None, (1, 1), "YX", None),
+        ((1, 1, 1), None, (1, 1, 1), "ZYX", None),
+        (
+            (1, 1, 1, 1),
+            None,
+            (1, 1, 1, 1),
+            "CZYX",
+            ["Channel:0"],
+        ),
+        (
+            (1, 1, 1, 1, 1),
+            None,
+            (1, 1, 1, 1, 1),
+            dimensions.DEFAULT_DIMENSION_ORDER,
+            ["Channel:0"],
+        ),
+        ((1, 1, 1), "AYX", (1, 1, 1), "AYX", None),
+        (
+            (1, 1, 1, 1),
+            "ZCYX",
+            (1, 1, 1, 1),
+            "ZCYX",
+            ["Channel:0"],
+        ),
+        (
+            (1, 1, 1, 1, 1),
+            "ABCYX",
```

Comment:
do you allow nonstandard dims when passing through the aicsimage interface?  I would think we'd reject any known_dims not in the TCZYX..  It would even be ok to do this for the reader for consistency, although not as important.

Reply:
The reader allows any dims, the `AICSImage` object will simply act as normal and only pull the "standard" `TCZYX`.

Reply:
how do I interpret "pull" here?  If I create a `AICSImage(myarray, known_dims="ATCZYX")` will it error out because of the A?

Reply:
Hmmmm I don't actually know. I should add a test for that in the `test_aicsimage` file.

Reply:
This is now resolved! And can be seen in tests: https://github.com/AllenCellModeling/aicsimageio/pull/197/files#diff-1e60d9c05612289c3ec21039f6471b986ed17da3bc3aba1e21409cc0f57c076eR448

Basically, I removed checking the known dims on the AICSImage object entirely as it's just an extra check that doesn't matter when we are just going to pull the standard dims back up from the reader. So this turns into: "let the base reader handle it" and it works nicely.